core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -150,44 +150,127 @@ describe('CreateActivityButtonComponent', () => {
+        return <NgbModalRef>({
+          result: Promise.resolve('success')
+        });
+      });
+      const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
 
-    expect(component.userIsLoggedIn).toBe(true);
-    expect(component.canCreateCollections).toBe(true);
-    expect(component.initCreationProcess).toHaveBeenCalled();
-  }));
+      component.initCreationProcess();
 
-  it('should create a new exploration automatically if the user cannot create' +
-    ' collections and is redirected to creator dashboard after' +
-    ' logging in', fakeAsync(() => {
-    spyOn(userService, 'getUserInfoAsync')
-      .and.returnValue(Promise.resolve(userInfoForNonCollectionCreator));
-    const explorationCreationServiceSpy = spyOn(
-      explorationCreationService, 'createNewExploration');
+      expect(replaceSpy).not.toHaveBeenCalled();
+      expect(modalSpy).toHaveBeenCalled();
+    });
 
-    expect(component.userIsLoggedIn).toBe(false);
-    expect(component.canCreateCollections).toBe(false);
+    it('should handle modal\'s failure callback' +
+    ' and stop creation process', fakeAsync(() => {
+      component.creationInProgress = false;
+      component.canCreateCollections = true;
+      spyOn(urlService, 'getPathname').and.returnValue(
+        '/creator-dashboard');
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return <NgbModalRef>({
+          result: Promise.reject('cancel')
+        });
+      });
 
-    component.ngOnInit();
-    tick();
-    fixture.detectChanges();
+      component.initCreationProcess();
+      tick();
 
-    expect(component.userIsLoggedIn).toBe(true);
-    expect(component.canCreateCollections).toBe(false);
-    expect(explorationCreationServiceSpy).toHaveBeenCalled();
-  }));
+      expect(modalSpy).toHaveBeenCalled();
+      expect(component.creationInProgress).toBe(false);
+    }));
+  });
+
+  describe('when user cannot create collection', () => {
+    it('should create a new exploration automatically if' +
+      ' url parameter \'mode\' is set as \'create\'', fakeAsync(() => {
+      spyOn(userService, 'getUserInfoAsync')
+        .and.returnValue(Promise.resolve(userInfoForNonCollectionCreator));
+      const explorationCreationServiceSpy = spyOn(
+        explorationCreationService, 'createNewExploration');
+      spyOn(urlService, 'getUrlParams').and.returnValue({
+        mode: 'create'
+      });
+
+      expect(component.userIsLoggedIn).toBe(false);
+      expect(component.canCreateCollections).toBe(false);
+
+      component.ngOnInit();
+      tick();
+      fixture.detectChanges();
+
+      expect(component.userIsLoggedIn).toBe(true);
+      expect(component.canCreateCollections).toBe(false);
+      expect(urlService.getUrlParams).toHaveBeenCalled();
+      expect(explorationCreationServiceSpy).toHaveBeenCalled();
+    }));
+
+    it('should create new exploration if the user is on' +
+      ' the creator dashboard page', () => {
```

Comment:
Which part of this implementation guarantees that "the user is on the creator dashboard page"?